pkg/clusterdeletion/deletion.go

```diff
@@ -82,72 +87,91 @@ func (d *Deletion) CleanupCluster(ctx context.Context, log *zap.SugaredLogger, c
 
-	if shouldDeleteLBs {
-		deletedSomeLBs, err := d.cleanupLBs(ctx, log, cluster)
-		if err != nil {
-			return fmt.Errorf("failed to cleanup LBs: %w", err)
+		// We'll set this to true in case we deleted something. This is meant to requeue as long as all resources are really gone
+		// We'll use it for LB's and PV's as well, so the Kubernetes controller manager does the cleanup of all resources in parallel
+		var deletedSomeResource bool
+
+		if shouldDeleteLBs {
+			deletedSomeLBs, err := d.cleanupLBs(ctx, log, cluster)
+			if err != nil {
+				return fmt.Errorf("failed to cleanup LBs: %w", err)
+			}
+			deletedSomeResource = deletedSomeResource || deletedSomeLBs
 		}
-		deletedSomeResource = deletedSomeResource || deletedSomeLBs
-	}
 
-	if shouldDeletePVs {
-		deletedSomeVolumes, err := d.cleanupVolumes(ctx, cluster)
-		if err != nil {
-			return fmt.Errorf("failed to cleanup PVs: %w", err)
+		if shouldDeletePVs {
+			deletedSomeVolumes, err := d.cleanupVolumes(ctx, log, cluster)
+			if err != nil {
+				return fmt.Errorf("failed to cleanup PVs: %w", err)
+			}
+			deletedSomeResource = deletedSomeResource || deletedSomeVolumes
 		}
-		deletedSomeResource = deletedSomeResource || deletedSomeVolumes
-	}
 
-	// If we deleted something it is implied that there was still something left. Just return
-	// here so the finalizers stay, it will make the cluster controller requeue us after a delay
-	// This also means that we may end up issuing multiple DELETE calls against the same resource
-	// if cleaning up takes some time, but that shouldn't cause any harm
-	// We also need to return when something was deleted so the checkIfAllLoadbalancersAreGone
-	// call gets an updated version of the cluster from the lister
-	if deletedSomeResource {
-		return nil
-	}
+		// If we deleted something it is implied that there was still something left. Just return
+		// here so the finalizers stay, it will make the cluster controller requeue us after a delay
+		// This also means that we may end up issuing multiple DELETE calls against the same resource
+		// if cleaning up takes some time, but that shouldn't cause any harm
+		// We also need to return when something was deleted so the checkIfAllLoadbalancersAreGone
+		// call gets an updated version of the cluster from the lister
+		if deletedSomeResource {
+			return nil
+		}
 
-	lbsAreGone, err := d.checkIfAllLoadbalancersAreGone(ctx, cluster)
-	if err != nil {
-		return fmt.Errorf("failed to check if all Loadbalancers are gone: %w", err)
-	}
-	// Return so we check again later
-	if !lbsAreGone {
-		return nil
+		lbsAreGone, err := d.checkIfAllLoadbalancersAreGone(ctx, cluster)
+		if err != nil {
+			return fmt.Errorf("failed to check if all Loadbalancers are gone: %w", err)
+		}
+		// Return so we check again later
+		if !lbsAreGone {
+			return nil
+		}
 	}
 
+	d.recorder.Event(cluster, corev1.EventTypeNormal, "ClusterCleanup", "Cleanup of volumes/LoadBalancers has been completed.")
```

Comment:
Uber nit: The other event uses `PersistentVolumeClaims`, so the API resource type. Can we also adjust this to something like

```suggestion
	d.recorder.Event(cluster, corev1.EventTypeNormal, "ClusterCleanup", "Cleanup of LoadBalancers / PersistentVolumes has been completed.")
```

So the two events are more consistent?